keps/sig-instrumentation/3077-contextual-logging/kep.yaml

```diff
@@ -0,0 +1,39 @@
+title: Contextual logging
+kep-number: 3077
+authors:
+  - "@pohly"
+owning-sig: sig-instrumentation
+participating-sigs:
+  - sig-scheduling
+status: implementable
+creation-date: 2021-12-06
+reviewers:
```

Comment:
I think it would be good to get feedback from @dims (who oversaw the klog/v2 transition and understands ecosystem ripples) and @thockin (from a dependency / injection / logr perspective)

Reply:
@thockin looked at this already beginning of December (https://github.com/kubernetes/enhancements/pull/3078#pullrequestreview-825741848), but another pass through the updated proposal would be very welcome.

Reply:
@pohly @liggitt i can see us getting this to Alpha. After that it becomes very fuzzy unless we have automation tools that mechanically change things (to reduce reviewer fatigue, churn/rebases waiting for reviews etc). I can see the structured logging and this getting done together, but it may be better if we can at least build automation to go from code already migrated to structured logging to support contextual logging possibly. i feel that different parts of our code base will move at different speeds and bringing the entire community with us is a big problem as well. We first have to ensure things we use are converted first and have a tagged release etc. my brain hurts. I am ok to get this idea in, get to alpha and revisit.

Reply:
> automation tools that mechanically change things (to reduce reviewer fatigue, churn/rebases waiting for reviews etc)

That only helps the developers who prepare PRs. LGTM and approval would still have to be manual, right?

> I can see the structured logging and this getting done together, but it may be better if we can at least build automation to go from code already migrated to structured logging to support contextual logging possibly.

Such a tool should be feasible. I've converted 68 files under pkg/scheduler manually in roughly a day and most of that was work that a tool could do, at least in those cases where a context was already getting passed in. More work is needed when a new parameter is needed, that better gets left to a developer to decide and implement.